### PR TITLE
fix: filter out cancelled newsletters

### PR DESCRIPTION
**Pull request by cemms1**
## What does this change?

The newsletters API is going to start returning cancelled newsletters in the response due to a requirement to have access to this data. Some issues were found with this update where the [all newsletters page](https://m.code.dev-theguardian.com/email-newsletters) was displaying the cancelled newsletters so we need to prevent these from displaying before releasing to prod.

There is a similar PR in `identity` for this work: https://github.com/guardian/identity/pull/2132/files.

The reason we have to filter out cancelled newsletters on the "client" side of the "api" is that newsletters isn't a proper API and just serves a JSON from an s3 bucket through fastly. If/when newsletters API becomes a full API, we can add filtering params for this in the call itself but for now this is just to ensure that the change to include cancelled newsletters as default in newsletters does not affect other parts of the system.

## Does this change need to be reproduced in dotcom-rendering ?

- [x] No
- [ ] Yes (please indicate your plans for DCR Implementation)

## Screenshots

| Before      | After      |
|-------------|------------|
| ![before][] | ![after][] |
| ![before2][] | cancelled section no longer appears |

[before]: https://user-images.githubusercontent.com/43961396/180186111-8831b604-bcca-41a8-8732-fa1f54e035a7.png
[after]: https://user-images.githubusercontent.com/43961396/180186118-e7e5e033-c9b6-47ae-9a2e-f123c73f04cb.png
[before2]: https://user-images.githubusercontent.com/43961396/180186590-28ca1ff0-3d6d-43a5-9680-ea6278b127a2.png

### Does this affect other platforms?

- [ ] AMP 
- [ ] Apps
- [ ] Other (please specify)

### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

- [x] No
- [ ] Yes (please give details)

### Does this change break ad-free?

- [x] No
- [ ] It did, but tests caught it and I fixed it
- [ ] It did, but there was no test coverage so I added that then fixed it

### Does this change update the version of CAPI we're using?

- [x] No, all the existing database files are just fine
- [ ] Yes, and I have [re-run all the tests locally and checked in all the updated data/database/xyz files](https://github.com/guardian/frontend/blob/main/docs/03-dev-howtos/15-updating-test-database.md)

### Tested

- [x] Locally
- [x] On CODE (optional)

### CODE REVIEW

**Comment by dblatcher** [comment]
It's probably worth clarifying in the description that the "cancelled" newsletters aren't actually appearing on the PROD all newsletters page at the moment, but that they would appear if we went ahead with the changes to the API and its data source without making this fix.

**Review by dblatcher — APPROVED** [review]
This looks good to me (although my scala is limited)! It will need an approval from the dotcom team too before it can be merged.

**Review by sunnyclimber456 — APPROVED** [review]
Looks great! Goodbye cancelled newsletters!

**Review by mxdvl — APPROVED** [review]

**Pull request merged**
